BUILDING.md

```diff
@@ -759,7 +759,156 @@ as `deps/icu` (You'll have: `deps/icu/source/...`)
 
 ## Building Node.js with FIPS-compliant OpenSSL
 
-The current version of Node.js does not support FIPS.
+The current version of Node.js does not support FIPS when statically linking
+(the default) with OpenSSL 1.1.1 but for dynamically linking it is possible
+to enable FIPS using the configuration flag `--openssl-is-fips`.
+
+### Configuring and Building quictls/openssl for FIPS
+
+For quictls/openssl 3.0 it is possible to enable FIPS when dynamically linking.
+Node.js currently uses openssl-3.0.0-alpha15+quic which can be configured as
+follows:
+```console
+$ git clone git@github.com:quictls/openssl.git
+$ cd openssl
+$ ./config -Werror --strict-warnings --debug --prefix=/path/to/install/dir/ shared enable-fips linux-x86_64
+```
+This can be compiled using:
+```console
+$ make -j8
+$ make install_ssldirs
+```
+Next set the `PATH` environment variable to point to the `openssl` executable
+in the `apps` directory:
+```console
+$ export PATH=./apps:$PATH
+```
+Now that should be enough to run the target `install_fips` but there seems to
+be an error with alpha-15 in regards to this target (this should be fixed in
+alpha-16). This can be worked around by removing the last option of the openssl
+fipsinstall command so that is looks like this:
+<!--lint disable no-tabs-->
+```console
+install_fips: install_sw
+	@$(ECHO) "*** Installing FIPS module configuration"
+	@$(ECHO) "fipsinstall $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).cnf"
+	@openssl fipsinstall -module $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME) \
+		-out $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).cnf
+```
+<!--lint disable no-tabs-->
+With those changes it should be possible to run the `install_fips` target:
+```console
+$ make install_fips
+```
+
+After the FIPS module and configuration file have been installed by the above
+instructions we also need to update `/path/to/install/dir/ssl/openssl.cnf` to
+use the generated FIPS configuration file (`fips.so.cnf`):
+```text
+.include /path/to/install/dir/lib/ossl-modules/fips.so.cnf
```

Comment:
FWIW from openssl 3.0.0-alpha16+quic I appear to have `ssl/fipsmodule.cnf` and no config files under `lib`.

Reply:
Yep, this is a change from alpha-15 and I'll update this with the other changes/removals. Thanks